libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationNotificationProvider.kt

```diff
@@ -0,0 +1,38 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import android.content.Context
+import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation
+import com.mapbox.services.android.navigation.v5.navigation.notification.NavigationNotification
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress
+
+internal class NavigationNotificationProvider(applicationContext: Context, mapboxNavigation: MapboxNavigation) {
+
+    private var navigationNotification: NavigationNotification? = null
```

Comment:
Could `NavigationNotification` type non-nullable?

Reply:
I initially made it non-nullable. I think the right question is inside the `shutdown` method do we need to set `navigationNotification = null`? If yes, then this cannot be non-nullable

Reply:
I think the reason behind this was to make sure `NavigationNotification` was garbage collected (without leaking anything). Could we test and make sure we're not leaking? If so, we can make it non-nullable as you're saying.

Reply:
Since, this is a member variable, I would prefer to have it set to null. We cannot guarantee that this object will be garbage collected until the instance of `NavigationNotificationProvider` inside `NavigationService` is being garbage collected, which I don't see happening. I would leave it nullable for now. 
Once, we make sure that instance of `NavigationNotificationProvider` is being garbage collected we can make this member variable non-nullable at that point